### PR TITLE
Assignment 2 Choo Xiang Lun S2132552

### PR DESCRIPTION
**Pull request by XiangLun0713**
# Assignment 2
## The addressed component
The addressed ESLint problems in PR https://github.com/weihanyau/cytoscape-home/pull/9 span across multiple React components of the Cytoscape Home website. This includes the Base UI components (Button, Fields, Container, etc.), Feature components (Hero, Results, PrimaryFeatures), Layout components (Layout, AppScreen, BrowserFrame) and Wizard components (WizardDialog and related wizards).

## Dependency Analysis
I've created a Program Dependency Graph (PDG) to visualize component relationships and the scope of my changes. The reason I've chosen to draw a PDG for my changes is because my changes primarily involve fixing multiple React components across the project. Therefore, a PDG can visualize how these components are interconnected effectively and how my ESLint fixes affect the relationship between different modules. The graph is shown below: 
eslint_fix_impact_analysis_program_dependency_graph

The analysis reveals three critical components that serve as core dependencies throughout the application: `Button.jsx`, which is extensively used across feature components, `Layout.jsx`, which controls the main structure and component composition and `WizardDialog.jsx`, which is central to all wizard-based interactions. The graph was separated into four main component categories, with direct dependencies flowing from base components to feature implementations. 

The graph is **complete** as it covers all affected components (addressing all 137 errors and 4 warnings) and provide a clear visualization of inter-component relationships as well as the logical grouping of related components. It also shows the explicit indication of the specific fixes applied to each component.

## Impact Analysis and Insights
Through the dependency analysis, we discovered that prop-types validation issues were systematic across components as there are 112 prop-types validation issues out of the total 141 errors. This shows that the project is not type safe at the moment and we could rework the project to use TypeScript instead of Javascript for more robust prop validation practices in the codebase. 

Other than that, my graph highlighted three components as critical dependencies affecting multiple other components which are the Button, Layout, and WizardDialog components. Therefore, making them priority targets for maintenance efforts. 

In a nutshell, my ESLint fixes will significantly improve code maintainability and enhance the developer experience by providing better IDE support through proper prop type definitions.

### CODE REVIEW

**Comment by suhadaudd11** [comment]
good
